binding/c/example.c need to be updated

- libsql git commit version.
```
[me@fedora c]$ git log -1
commit 4de573727ccd65225b9d4944d49cb61b89196e87 (HEAD -> main, origin/main, origin/HEAD)
Merge: 69b0bbcd7 421922061
Author: ad hoc <postma.marin@protonmail.com>
Date:   Wed Nov 8 12:44:19 2023 +0000

    Merge pull request #588 from tursodatabase/conccurent-conns-test-timeout

    add timeout to conccurent connections test
[me@fedora c]$
```
- example.c compile error
```
[me@fedora c]$ sh -x compile.sh
+ gcc -I include example.c ../../target/debug/libsql_experimental.a ../../../.libs/libsqlite3.a
example.c: In function ‘main’:
example.c:13:14: error: too few arguments to function ‘libsql_open_ext’
   13 |         db = libsql_open_ext(":memory:");
      |              ^~~~~~~~~~~~~~~
In file included from example.c:3:
include/libsql.h:43:5: note: declared here
   43 | int libsql_open_ext(const char *url, libsql_database_t *out_db, const char **out_err_msg);
      |     ^~~~~~~~~~~~~~~
example.c:17:16: error: too few arguments to function ‘libsql_connect’
   17 |         conn = libsql_connect(db);
      |                ^~~~~~~~~~~~~~
include/libsql.h:47:5: note: declared here
   47 | int libsql_connect(libsql_database_t db, libsql_connection_t *out_conn, const char **out_err_msg);
      |     ^~~~~~~~~~~~~~
example.c:21:16: error: too few arguments to function ‘libsql_execute’
   21 |         rows = libsql_execute(conn, "SELECT 1");
      |                ^~~~~~~~~~~~~~
include/libsql.h:51:5: note: declared here
   51 | int libsql_execute(libsql_connection_t conn, const char *sql, libsql_rows_t *out_rows, const char **out_        err_msg);
      |     ^~~~~~~~~~~~~~
example.c:25:33: warning: implicit declaration of function ‘libsql_row_count’; did you mean ‘libsql_column_count        ’? [-Wimplicit-function-declaration]
   25 |         for (int row = 0; row < libsql_row_count(rows); row++) {
      |                                 ^~~~~~~~~~~~~~~~
      |                                 libsql_column_count
example.c:30:45: warning: implicit declaration of function ‘libsql_value_text’; did you mean ‘libsql_open_ext’?         [-Wimplicit-function-declaration]
   30 |                         const char *value = libsql_value_text(rows, row, col);
      |                                             ^~~~~~~~~~~~~~~~~
      |                                             libsql_open_ext
example.c:30:45: warning: initialization of ‘const char *’ from ‘int’ makes pointer from integer without a cast         [-Wint-conversion]
[me@fedora c]$

```
``

--- a/.github/workflows/c-bindings.yml
+++ b/.github/workflows/c-bindings.yml
@@ -1,0 +1,40 @@
+name: C bindings tests CI
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+  merge_group:
+    branches: [ "main" ]
+
+jobs:
+  c-bindings:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - uses: dtolnay/rust-toolchain@stable
+
+    - name: Install Protoc
+      uses: arduino/setup-protoc@v2
+
+    - name: Set up cargo cache
+      uses: actions/cache@v3
+      continue-on-error: false
+      with:
+        path: |
+          ~/.cargo/bin/
+          ~/.cargo/registry/index/
+          ~/.cargo/registry/cache/
+          ~/.cargo/git/db/
+          target/
+        key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+        restore-keys: ${{ runner.os }}-cargo-
+
+    - name: Build crates
+      run: cargo build
+
+    - name: Build C bindings example
+      working-directory: bindings/c
+      run: make

--- a/bindings/c/Makefile
+++ b/bindings/c/Makefile
@@ -1,0 +1,13 @@
+OS := $(shell uname)
+CFLAGS := -Iinclude
+LDFLAGS := -lm
+
+ifeq ($(OS),Darwin)
+	CFLAGS += -framework Security -framework CoreServices
+endif
+
+.PHONY: all
+all: example
+
+example: example.c
+	$(CC) -o $@ $(CFLAGS) $< ../../target/debug/libsql_experimental.a $(LDFLAGS)

--- a/bindings/c/example.c
+++ b/bindings/c/example.c
@@ -1,5 +1,3 @@
-/* gcc -I include example.c ../../target/debug/libsql_experimental.a ../../../.libs/libsqlite3.a && ./a.out */
-
 #include "libsql.h"
 #include <assert.h>
 #include <stdio.h>
@@ -8,30 +6,59 @@ int main(int argc, char *argv[])
 {
 	libsql_connection_t conn;
 	libsql_rows_t rows;
+	libsql_row_t row;
 	libsql_database_t db;
+	const char *err = NULL;
+	int retval = 0;
 
-	db = libsql_open_ext(":memory:");
-	if (!db) {
-		assert(0);
+	retval = libsql_open_ext(":memory:", &db, &err);
+	if (retval != 0) {
+		fprintf(stderr, "%s\n", err);
+		goto quit;
 	}
-	conn = libsql_connect(db);
-	if (!conn) {
-		assert(0);
+
+	retval = libsql_connect(db, &conn, &err);
+	if (retval != 0) {
+		fprintf(stderr, "%s\n", err);
+		goto quit;
 	}
-	rows = libsql_execute(conn, "SELECT 1");
-	if (!rows) {
-		assert(0);
+
+	retval = libsql_execute(conn, "SELECT 1", &rows, &err);
+	if (retval != 0) {
+		fprintf(stderr, "%s\n", err);
+		goto quit;
 	}
-	for (int row = 0; row < libsql_row_count(rows); row++) {
-		for (int col = 0; col < libsql_column_count(rows); col++) {
+
+	int num_cols = libsql_column_count(rows);
+
+	while ((retval = libsql_next_row(rows, &row, &err)) == 0) {
+		if (!err && !row) {
+			break;
+		}
+		for (int col = 0; col < num_cols; col++) {
 			if (col > 0) {
 				printf(", ");
 			}
-			const char *value = libsql_value_text(rows, row, col);
-			printf("%s", value);
+			long long value;
+			retval = libsql_get_int(row, col, &value, &err);
+			if (retval != 0) {
+				fprintf(stderr, "%s\n", err);
+			} else {
+				printf("%lld\n", value);
+			}
 		}
+		err = NULL;
 	}
+
+	if (retval != 0) {
+		fprintf(stderr, "%s\n", err);
+		goto quit;
+	}
+
+quit:
 	libsql_free_rows(rows);
 	libsql_disconnect(conn);
 	libsql_close(db);
+
+	return retval;
 }

